### PR TITLE
fix bugs in test and some conflic with FFMTextIterator

### PR DESCRIPTION
**Pull request by danyang-liu**
### Description

fix test about deeprec, ensure all the test about deeprec passed

some previous test failed due to the code changed, fix them.

### Related Issues

https://github.com/microsoft/recommenders/issues/1132

### Checklist:

- [yes] I have followed the [contribution guidelines](../CONTRIBUTING.md) and code style for this project.
- [yes] I have added tests covering my contributions.
- [yes] I have updated the documentation accordingly.
- [yes] This PR is being made to `staging` and not `master`.

### CODE REVIEW

**Comment by Leavingseason** [comment]
@miguelgfierro fig bugs

File: tests/smoke/test_notebooks_gpu.py [hunk 1]
```diff
@@ -119,8 +119,8 @@ def test_dkn_smoke(notebooks):
     )
     results = pm.read_notebook(OUTPUT_NOTEBOOK).dataframe.set_index("name")["value"]
 
-    assert results["res"]["auc"] == pytest.approx(0.4707, rel=TOL, abs=ABS_TOL)
-    assert results["res"]["acc"] == pytest.approx(0.5725, rel=TOL, abs=ABS_TOL)
+    assert results["res"]["auc"] == pytest.approx(0.5653, rel=TOL, abs=ABS_TOL)
+    assert results["res"]["mean_mrr"] == pytest.approx(0.1629, rel=TOL, abs=ABS_TOL)
```
Comment (miguelgfierro) [inline tests/smoke/test_notebooks_gpu.py:123]: I think we should remove the line of f1 and add 'ndcg@5', 'ndcg@10'

Comment (danyang-liu) [reply tests/smoke/test_notebooks_gpu.py:123]: ok, let me do it

Comment (danyang-liu) [reply tests/smoke/test_notebooks_gpu.py:123]: finished. @miguelgfierro

**Comment by danyang-liu** [comment]
@miguelgfierro I think we can merge this PR

**Comment by danyang-liu** [comment]
@miguelgfierro we changed data again, I am updating test, please wait a min, do not merge

**Comment by danyang-liu** [comment]
@miguelgfierro you can merge it, thanks!

**Pull request merged**